### PR TITLE
SSCS-10801 - add PO fields, Action Postponement request, view hearing date and hearing venue

### PR DESCRIPTION
**Pull request by hajam09**
### JIRA link (if applicable) ###
[SSCS-10801](https://tools.hmcts.net/jira/browse/SSCS-10801)[,](https://github.com/hmcts/sscs-ccd-definitions/pull/1016/commits/4e3fe314dc19649e63a11851f8b34e68aac83278) [SSCS-10814](https://tools.hmcts.net/jira/browse/SSCS-10814)

### Change description ###
add PO fields, Action Postponement request, view hearing date and hearing venue

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

### CODE REVIEW

**Comment by hajam09** [comment]
The work done in this branch fulfils: SSCS-10801, SSCS-10789 and SSCS-10814

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   {"LiveFrom": "01/01/2018", "CaseTypeID": "Benefit", "CaseEventID": "createTestCase", "UserRole": "caseworker-sscs", "CRUD": "R"},
   {"LiveFrom": "22/10/2019", "CaseTypeID": "Benefit", "CaseEventID": "postponementRequest", "UserRole": "caseworker-sscs-dwpresponsewriter", "CRUD": "CRUD"},
   {"LiveFrom": "22/10/2019", "CaseTypeID": "Benefit", "CaseEventID": "postponementRequest", "UserRole": "caseworker-sscs-superuser", "CRUD": "CRUD"},
+  {"LiveFrom": "22/10/2019", "CaseTypeID": "Benefit", "CaseEventID": "actionPostponementRequest", "UserRole": "caseworker-sscs-judge", "CRUD": "CRUD"},
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:12]: I don't see any mentions on the ticket about this

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 10},
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-registrar", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 12},
   {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-superuser", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 18},
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-dwpresponsewriter", "CaseFieldID": "workBasketHearingDate", "Label": "Hearing Date", "DisplayOrder": 19},
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-superuser", "CaseFieldID": "workBasketHearingDate", "Label": "Hearing Date", "DisplayOrder": 19},
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "workBasketHearingDate", "Label": "Hearing Date", "DisplayOrder": 19},
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:9]: Are these changes required too?

Comment (dattipoeHMCTS) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:9]: (and below)

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]
I must admit I haven't touched work basket fields. Have you confirmed this works locally? You might need to add some more entries in AuthorisationCaseField. Let me know please.

You also need to bump version numbers on CCD PRs: https://tools.hmcts.net/confluence/display/SSCS/CCD+Definitions+Info

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json (hunk 2 above)
Comment (hajam09) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:9]: Made changes to complete SSCS-10814

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 1 above)
Comment (hajam09) [reply benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:12]: [SSCS-10789](https://tools.hmcts.net/jira/browse/SSCS-10789)

File: benefit/data/sheets/WorkBasketResultFields/WorkBasketResultFields-sandl-nonprod.json [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "workBasketHearingDate", "Label": "Hearing Date", "DisplayOrder": 18},
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-dwpresponsewriter", "CaseFieldID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "DisplayOrder": 19},
-  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-superuser", "CaseFieldID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "DisplayOrder": 19}
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-superuser", "CaseFieldID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "DisplayOrder": 19},
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "DisplayOrder": 19},
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "DisplayOrder": 19}
```
Comment (hajam09) [inline benefit/data/sheets/WorkBasketResultFields/WorkBasketResultFields-sandl-nonprod.json:14]: Do I need to add an entry like this on SearchResultFields-sandl-nonprod.json?

**Comment by hajam09** [comment]
When I logged in as a judge and clerk the filed was visible in result bar and in the filters.

File: benefit/data/sheets/WorkBasketResultFields/WorkBasketResultFields-sandl-nonprod.json (hunk 3 above)
Comment (dattipoeHMCTS) [reply benefit/data/sheets/WorkBasketResultFields/WorkBasketResultFields-sandl-nonprod.json:14]: For the filters to appear, I think so

File: benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json [hunk 4]
```diff
@@ -1256,6 +1256,8 @@
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "CaseFieldID": "workBasketHearingDate", "UserRole": "citizen", "CRUD": "CRUD"},
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "CaseFieldID": "workBasketHearingEpimsId", "UserRole": "caseworker-sscs-superuser", "CRUD": "CRUD"},
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "CaseFieldID": "workBasketHearingEpimsId", "UserRole": "caseworker-sscs-systemupdate", "CRUD": "CRUD"},
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "CaseFieldID": "workBasketHearingEpimsId", "UserRole": "caseworker-sscs-clerk", "CRUD": "CRUD"},
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json:1259]: I'm not sure why the others are in the prod sheets, could you confirm whether this should be in as part of the S&L release?

File: benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json [hunk 5]
```diff
@@ -1256,6 +1256,8 @@
...
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "CaseFieldID": "workBasketHearingEpimsId", "UserRole": "caseworker-sscs-judge", "CRUD": "CRUD"},
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json:1260]: You might need to add this for caseworker-sscs as well

**Comment by dattipoeHMCTS** [comment]
Can I confirm whether this has been tested locally? Would be good if you could give business a quick demo locally

File: benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json (hunk 4 above)
Comment (hajam09) [reply benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json:1259]: Doesn't the PET team make changes directly to prod?

**Review by dattipoeHMCTS — APPROVED** [review]
Approved after successful demo

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
Let's review the "prod" items again before approving

File: benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json (hunk 4 above)
Comment (filippo-possenti-version1) [reply benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json:1259]: After discussion, this should be a nonprod change, so this change will be reverted and applied to the relevant nonprod sheet

File: benefit/data/sheets/CaseType/CaseType.json [hunk 6]
```diff
@@ -1 +1 @@
-[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.0_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
+[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.0_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
```
Comment (filippo-possenti-version1) [inline benefit/data/sheets/CaseType/CaseType.json:1]: After discussion, this should be a nonprod change, so this change will be reverted and applied to the relevant nonprod sheet

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]
Needs a version bump

**Review by filippo-possenti-version1 — DISMISSED** [review]
Approving my own changes, which is just bumping up a version number.

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 1 above)
Comment (filippo-possenti-version1) [reply benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json]: Ideally one PR should address one ticket, but let's get this one out of the way. I'll mark this conversation as resolved.

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json (hunk 2 above)
Comment (filippo-possenti-version1) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:9]: Ideally one PR should address one ticket, but let's get this one out of the way. I'll mark this conversation as resolved.

File: benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json (hunk 4 above)
Comment (filippo-possenti-version1) [reply benefit/data/sheets/AuthorisationCaseField/AuthorisationCaseField.json]: Relevant changes applied. Resolving this conversation.

File: benefit/data/sheets/CaseType/CaseType.json (hunk 6 above)
Comment (filippo-possenti-version1) [reply benefit/data/sheets/CaseType/CaseType.json]: Relevant changes applied. Resolving this conversation.

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]
Could you please generate a local prod version and check that you can still create/update a case?

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by ss-evoco — APPROVED** [review]

File: benefit/VERSION.yaml [hunk 7]
```diff
@@ -1 +1 @@
-TAG: 5.15.2
+TAG: 5.15.3
```
Comment (mcareyma) [inline benefit/VERSION.yaml:1]: I think I am grabbing that value in my PR, so which ever of us gets approval second will have to move it further.

File: benefit/data/sheets/CaseType/CaseType.json [hunk 8]
```diff
@@ -1 +1 @@
-[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.2_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
+[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.3_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
```
Comment (mcareyma) [inline benefit/data/sheets/CaseType/CaseType.json:1]: likewise with this value

**Review by mcareyma — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]
